test/caPutJsonLogTest.cpp

```diff
@@ -692,6 +716,18 @@ void testDbf(const char *pv, chtype type,
     }
 }
 
+void testMetadataHelper(std::map<std::string, std::string> metadata )
+{
+    char desc1[] = "Description 1";
+    char desc2[] = "Description 2";
+    std::map<std::string, std::string>::iterator meta_it;
+    for(meta_it = metadata.begin(); meta_it != metadata.end(); meta_it++){
+        logger->addMetadata(meta_it->first,meta_it->second);
+    }
+    testDbf<dbr_string_t>("longout_DBF_STRING.DESC", DBR_STRING, desc1, 1, desc2, 1, "Metadata test");
```

Comment:
Can you explain this? I don' t understand this testDbf call.

Reply:
It's the call that sends the json message the message via ca_array_put and waits on the log server for the message and proceed to do the parsing. I did the test only with one datatype(string) since there are tests that already cover all the datatypes.